PLOS-Formulas/lagotto-formula
# Alm 1045 dockerize

https://jira.plos.org/jira/browse/ALM-1045

## Related PRs:

https://github.com/PLOS/lagotto/pull/40
https://github.com/PLOS/molten/pull/1256

**Note** that this PR merges into a branch called `aws` instead of dev. This PR breaks compatibility with trusty--I felt like it was too complicated and low value to maintain this in Soma. The `aws` branch is meant to be the basis of what we'll create `PLOS/lagotto-fomula` from (heads up @heschlie).


## What's here?

* Run puma inside a container
* Run nginx inside a container
* Reworked how map.jinja is translated into ENV vars for the container
* Got rid of a lot of stuff that was rendered unnecessary by containers
* Set up consul service discovery
* Moved db seeding concerns from these states into the lagotto repo. Seeding no longer happens automatically (it seems like it's best done manually)

## What Consul services are advertised now?
( https://confluence.plos.org/confluence/display/SRE/Service+Naming+Convention )

* alm-manager-memcache
* alm-manager-sql
* alm-manager-nosql (kinda, this couch is only managed by these salt states in monolithic environments)
* alm-manager-redis
* alm-manager-app
* alm-manager-web

## How to test this

* check out this molten PR: https://github.com/PLOS/molten/pull/1256
* change the image tag name at https://github.com/PLOS-Formulas/lagotto-formula/pull/19/files#diff-bc9ee8db26e65f08826be6c6cd1009dfR49 from `develop` to `ALM-1045-dockerize` (Once https://github.com/PLOS/lagotto/pull/40 merges, this change will be unnecessary)
* `vagrant up` from `molten/vagrant/apps/lagotto`
* You'll see twenty-something errors (from what I can tell, these are race conditions arising from docker and consul setup)
* `vagrant provision` (should run clean)
* check it out at http://lagotto.vagrant.local
